main.tf

```
@@ -1,20 +1,56 @@
-resource "scaleway_object_bucket" "this" {
+resource "scaleway_object_bucket" "main" {
   force_destroy       = var.force_destroy
   name                = var.name
   object_lock_enabled = var.versioning_enabled
 
   region     = var.region
   project_id = var.project_id
 
+  tags = var.tags == null ? {} : {
+     for tag in var.tags :
+     tag => tag
+  }
+
   versioning {
     enabled = var.versioning_enabled
   }
+
+  dynamic "lifecycle_rule" {
+    for_each = var.lifecycle_rule
+    iterator = terraform_rules
+    content {
+      id = terraform_rules.key
+      enabled = lookup(terraform_rules.value, "enabled", "true")
+      prefix = lookup(terraform_rules.value, "prefix", null)
+      abort_incomplete_multipart_upload_days = lookup(terraform_rules.value, "abort_incomplete_multipart_upload_days", null)
+
+      tags = lookup(terraform_rules.value, "tags", null) == null ? {} : {
+         for key, value in terraform_rules.value.tags :
+         key => value
+      }
+
+      dynamic "expiration" {
+        for_each = length(lookup(terraform_rules.value, "expiration", "")) == 0 ? [] : [terraform_rules.value.expiration]
+        content {
+            days = terraform_rules.value.expiration
+        }
+      }
+
+      dynamic "transition" {
+        for_each = length(lookup(terraform_rules.value, "transition", "")) == 0 || length(lookup(terraform_rules.value, "storage_class", "")) == 0 ? [] : [terraform_rules.value.transition]
+        content {
+          days          = terraform_rules.value.transition
+          storage_class = terraform_rules.value.storage_class
+        }
+      }
+    }
+  }
 }
 
-resource "scaleway_object_bucket_lock_configuration" "this" {
+resource "scaleway_object_bucket_lock_configuration" "main" {
```

Same than before: Add a "moved" block. Should be:
```suggestion
moved {
  from = scaleway_object_bucket_lock_configuration.this
  to   = scaleway_object_bucket_lock_configuration.main
}

resource "scaleway_object_bucket_lock_configuration" "main" {
```